be/src/storage/tablet_schema.h

```diff
@@ -40,35 +40,95 @@ class SegmentReaderWriterTest_TestStringDict_Test;
 } // namespace segment_v2
 
 class TabletColumn {
+    struct ExtraFields {
+        std::string default_value;
+        std::vector<TabletColumn> sub_columns;
+        bool has_default_value = false;
+    };
+
 public:
+    // To developers: if you changed the typedefs, don't forget to reorder class members to
+    // minimize the memory space of TabletColumn, i.e, sizeof(TabletColumn)
+    typedef int32_t unique_id_type;
+    typedef int32_t length_type;
+    typedef uint8_t index_length_type;
+    typedef uint8_t precision_type;
+    typedef uint8_t scale_type;
```

Comment:
1. prefer to use `using` syntax.
2. Type's name need to be in camel case. Prefer ScaleType to scale_type

Reply:
updated, PTAL